src/platform/forms-system/src/js/components/FormSignature.jsx

```diff
@@ -41,8 +41,11 @@ export const FormSignature = ({
   onSectionComplete,
 }) => {
   // Input states
-  const [signature, setSignature] = useState({ value: '', dirty: false });
-  const [checked, setChecked] = useState(false);
+  const [signature, setSignature] = useState({
+    value: formData?.signature ?? '', // Pre-populate with existing signature if available
+    dirty: formData?.signature?.length > 0, // will be dirty if any prev signature is present
+  });
+  const [checked, setChecked] = useState(formData?.AGREED ?? false);
```

Comment:
Main issue is that the component state was being discarded in re-render on submit (component's position in the render tree changing for some reason?). 

This fixes the issue by `useState`ing any existing signature data when the component is initialized. If none is present just behaves the way it did before.